### PR TITLE
Move examples to sphinx-gallery

### PR DESCRIPTION
**Pull request by banesullivan**
These changes moves all of the examples to work with [Sphinx Gallery](https://sphinx-gallery.github.io/stable/index.html) (SG). SG provides a way to show off your examples with really nice formatting and incorporating the examples as test cases to make sure the example in your docs are always in working state when a user tries them out.

Previously, the examples were in two places, in scripts under the `examples` directory and in code blocks within the documentation pages. Having the examples in two places can lead to discrepancies and maintenance issues, these changes consolidate the examples into a single place (doc and code) and provide a way to test all of the examples.

-----

I would highly recommend that you all switch from readthedocs.org to Travis and GitHub pages for building and deploying your docs. Readthedocs build environment is very limited and you will not be able to showcase any 3D examples with PyVista if you stay on readthedocs. Also, the whole purpose of this is to have your examples all in one place where they can be tested and incorporated into the documentation - Sphinx-gallery will run each example when building the docs and that is a great build status check to incorporate in with GitHub status checks (readthedocs does not support that)

### CODE REVIEW

**Comment by banesullivan** [comment]
In it's current state, everything should build on readthedocs fine, but if you want to show add the example in #36, then you'll have to switch to using GitHub pages with a service like [`doctr`](https://github.com/drdoctr/doctr)

**Comment by banesullivan** [comment]
Also, it looks like there are a ton of images in the `docs/source/pics` directory... most of these are now automatically generated by sphinx-gallery during the docs build, so they can be deleted.

**Comment by banesullivan** [comment]
It appears the builds are failing because you all upload to PyPI on every build (and I don't have privileges to do that)?

**Comment by banesullivan** [comment]
the base has been changed to `develop`

File: docs/source/conf.py [hunk 1]
```diff
@@ -247,3 +248,47 @@ def setup(app):
...
+        "examples/00_intro/",
+        "examples/01_random_field/",
+        "examples/02_cov_model/",
+        "examples/03_variogram/",
+        "examples/04_vector_field/",
+        "examples/05_kriging/",
+        "examples/06_conditioned_fields/",
+        "examples/07_transformations/",
+    ],
+    # Patter to search for example files
```
Comment (MuellerSeb) [inline docs/source/conf.py]: Typo: "Pattern"

File: docs/source/conf.py [hunk 2]
```diff
@@ -247,3 +248,47 @@ def setup(app):
...
+    "filename_pattern": r"\.py",
+    # Remove the "Download all examples" button from the top level gallery
+    "download_all_examples": False,
+    # Sort gallery example by file name instead of number of lines (default)
+    "within_subsection_order": FileNameSortKey,
+    # directory where function granular galleries are stored
+    "backreferences_dir": None,
+    # Modules for which function level galleries are created.  In
+    "doc_module": "gstools",
+    #"image_scrapers": ('pyvista', 'matplotlib'),
```
Comment (MuellerSeb) [inline docs/source/conf.py]: Why these comments?

File: docs/source/tutorials.rst [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 =================
 
 In the following you will find several Tutorials on how to use GSTools to
 explore its whole beauty and power.
 
+
 .. toctree::
-   :maxdepth: 1
+   :includehidden:
+   :maxdepth: 3
```
Comment (MuellerSeb) [inline docs/source/tutorials.rst]: Why ``:maxdepth: 3``? We think 1 is enough, as the chapters are very short in general.

File: docs/source/conf.py (hunk 2 above)
Comment (banesullivan) [reply docs/source/conf.py]: For *when* you all start using PyVista for some 3D viz in the example gallery, you will need to add the pyvista image scraper. I figured I would leave this here for when that happens

**Pull request closed without merging**